Task2/Project/Shapes/Commands/Vertex/ChangeLocation.cs

```diff
@@ -1,23 +1,55 @@
-﻿namespace Shapes.Commands.Vertex
+namespace Shapes.Commands.Vertex
 {
+    /// <summary>
+    /// Represents command for <see cref="Models.Vertex"/> that changes its location.
+    /// </summary>
     public class ChangeLocation : Interfaces.ICommand
     {
-        public string Name
+        // FIELDS
+        private Models.Vertex vertex;
+        private System.Windows.Point location;
+        private System.Windows.Point prevLocation;
+        // PROPERTIES
+        /// <summary>
+        /// Command name.
+        /// </summary>
+        public string Name => "Vertex location changed";
+        // CONSTRUCTORS
+        /// <summary>
+        /// Constructor with parameters.
+        /// </summary>
+        /// <param name="vertex">Current vertex.</param>
+        /// <param name="location">New location.</param>
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown when vertex is null.
+        /// </exception> 
+        public ChangeLocation(Models.Vertex vertex, System.Windows.Point location)
         {
-            get
+            if (vertex != null)
             {
-                throw new System.NotImplementedException();
+                this.vertex = vertex;
+                this.location = location;
+            }
+            else
+            {
+                throw new System.ArgumentNullException("Vertex is null.");
             }
         }
-
+        // METHODS
+        /// <summary>
+        /// Changes <see cref="Models.Vertex"/>'s location.
+        /// </summary>
         public void Execute()
         {
-            throw new System.NotImplementedException();
+            prevLocation = vertex.Location;

```

Comment:
should be in constructor, since its initialize logic, not the lofic of Execute method

Reply:
Done